src/nuget/Microsoft.DotNet.Build.Tasks.Feed.nuspec

```diff
@@ -20,11 +20,11 @@
     </dependencies>
   </metadata>
   <files>
-    <file src="Microsoft.DotNet.Build.Tasks.Feed\Microsoft.DotNet.Build.Tasks.Feed.dll" target="lib\netstandard1.5" />
-    <file src="Microsoft.DotNet.Build.Tasks.Feed.net45\*.dll" target="lib\netstandard1.5" Exclude="Microsoft.DotNet.Build.Tasks.Feed\Microsoft.DotNet.Build.Tasks.Feed.dll"/>
-    <file src="Microsoft.DotNet.Build.Tasks.Feed.net45\*.dll" target="lib\net46" />
-    <file src="Microsoft.DotNet.Build.Tasks.Feed\PackageFiles\**\*.*" target="lib\netstandard1.5" />
-    <file src="Microsoft.DotNet.Build.Tasks.Feed\PackageFiles\**\*.*" target="lib\net46" />
+    <file src="Microsoft.DotNet.Build.Tasks.Feed\Microsoft.DotNet.Build.Tasks.Feed.dll" target="build\netstandard1.5" />
+    <file src="Microsoft.DotNet.Build.Tasks.Feed.net45\*.dll" target="build\netstandard1.5" Exclude="Microsoft.DotNet.Build.Tasks.Feed\Microsoft.DotNet.Build.Tasks.Feed.dll"/>
+    <file src="Microsoft.DotNet.Build.Tasks.Feed.net45\*.dll" target="build\net46" />
+    <file src="Microsoft.DotNet.Build.Tasks.Feed\PackageFiles\*.props" target="build\netstandard1.5" />
```

Comment:
While I think this is fine I'm curious to understand what your plan is for importing the correct props file in BuildTools based repos?

Reply:
Isn't the plan to make a project that has a reference to this package?

Reply:
that project would be init-tools.msbuild and it would be used to publish but there isn't necessarily a clear way to do the import in the other projects like publish.proj. I think may need to figure out something based on BuildToolsTaskDir perhaps.

Reply:
I was thinking we move towards the model of treating it like an actual package reference. Have the repo's publish.proj reference it, restore for the correct tfm and auto import the appropriate task based on the framework it restore for.

Reply:
The current model seems to be treating it like a zip file, and unpacking it into a directory, which makes things more confusing.

Reply:
I don't really want a bunch of random projects in corefx (outside of external and init-tools) to have PackageReferences and require being restored. Instead these tools packages I want them to be restored and published via init-tools.msbuild and it will need to produce the correct layout under the Tools directory so that the assumptions of BuildTools still holds.

Reply:
My interpretation of the model that Wes is proposing...

Have a Common.props file like...

```
<Project ToolsVersion="12.0" DefaultTargets="PublishOutputLeg" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <PropertyGroup>
    <RunningOnCore Condition="'$(MSBuildRuntimeType)' == 'Core'">true</RunningOnCore>
    <_TaskDir>$(MSBuildThisFileDirectory)../lib/net46/</_TaskDir>
    <_TaskDir Condition="'$(RunningOnCore)' == 'true'">$(MSBuildThisFileDirectory)../lib/netstandard1.5/</_TaskDir>
  </PropertyGroup> 
 </Project>
```

The nuspec looks like...
```
  <files>
    <file src="Microsoft.DotNet.Build.Tasks.Feed\Microsoft.DotNet.Build.Tasks.Feed.dll" target="lib\netstandard1.5" />
    <file src="Microsoft.DotNet.Build.Tasks.Feed.net45\*.dll" target="lib\netstandard1.5" Exclude="Microsoft.DotNet.Build.Tasks.Feed\Microsoft.DotNet.Build.Tasks.Feed.dll"/>
    <file src="Microsoft.DotNet.Build.Tasks.Feed.net45\*.dll" target="lib\net46" />
    <file src="Microsoft.DotNet.Build.Tasks.Feed\PackageFiles\**\*.*" target="build" />
    <file src="SomeCommonLocation\Common.props" target="build" />
    <file src="..\obj\version.txt" target="" />
  </files>
```

The Publish.targets and Restore.targets include...
```
  <Import Project="$(MSBuildThisFileDirectory)Common.props" />
  <UsingTask TaskName="PushToBlobFeed" AssemblyFile="$(_TaskDir)Microsoft.DotNet.Build.Tasks.Feed.dll"/>
```
Then the user just does either `msbuild Publish.targets blah` or `dotnet msbuild Publish.targets blah` and things just work.

Regarding Common.props: I guess there's no consistency (currently), for what desktop or core framework we are targeting (ie, lib/net45 or lib/net46 for desktop; lib/netstandard13 or lib/netstandard1.5 or something else for core), but initially lib/net46 for desktop and lib/netstandard1.5 for core would work.  I could imagine a slightly adjusted model where each package can explicitly provide its desktop or core target frameworks.

Reply:
Yes that is close to what I'm expecting. I would put the UsingTask in the targets file in the package as well. 

Given our Tools layout though I'm not certain the relative paths you specified will work after we do the publish into the Tools directory. However I think we can figure out how to make those work based on the Tools layout as well as the package layout.  If not then perhaps a targets file in each TFM directly like @karajas has and then we could in a common place have select the correct import.